tests/services/base/test_client_factory.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+
+from orca.errors import ClientRequestError, ConfigError
+from orca.services.base import BaseConfig
+
+
+def test_that_config_is_set(client_factory):
+    assert hasattr(client_factory, "config")
+    assert isinstance(client_factory.config, BaseConfig)
+
+
+def test_that_client_class_is_set(service):
+    client_factory_cls = service["client_factory"]
+    assert hasattr(client_factory_cls, "client_class")
+
+
+def test_that_client_factory_can_generate_client(client_factory):
+    client = client_factory.create_client()
+    assert isinstance(client, client_factory.client_class)
+
+
+def test_for_an_error_when_testing_the_client_without_auth(client_factory):
+    client = client_factory.create_client()
+    with pytest.raises(ClientRequestError):
```

Comment:
Nit: Did we want to do light matching of the error message?

Reply:
IMHO, I don't think we should test for error messages because that's not formally part of the interface. In the docstrings, we indicate that an exception class will be raised if a certain situation is encountered. That's part of the interface. I think of the error message as more of an implementation detail.